core/src/main/java/org/apache/iceberg/BaseRewriteManifests.java

```diff
@@ -113,65 +119,98 @@ public RewriteManifests rewriteIf(Predicate<ManifestFile> pred) {
   }
 
   @Override
-  public List<ManifestFile> apply(TableMetadata base) {
-    Preconditions.checkNotNull(clusterByFunc, "clusterBy function cannot be null");
+  public RewriteManifests deleteManifest(ManifestFile manifest) {
+    deletedManifests.add(manifest);
+    return this;
+  }
+
+  @Override
+  public RewriteManifests addManifest(ManifestFile manifest) {
+    // the manifest must be rewritten with this update's snapshot ID
+    Map<Integer, PartitionSpec> specsById = ops.current().specsById();
+    try (ManifestReader reader = ManifestReader.read(ops.io().newInputFile(manifest.path()), specsById)) {
+      Set<ManifestEntry.Status> allowedStatuses = Sets.newHashSet(ManifestEntry.Status.EXISTING);
```

Comment:
Nit: this could be a static final constant.

Reply:
Done